src/control/server/config/server_test.go

```diff
@@ -236,53 +236,72 @@ func TestServerConfig_Constructed(t *testing.T) {
-					storage.NewTierConfig().
-						WithBdevClass("nvme").
-						WithBdevDeviceList("0000:81:00.0"),
-				).
-				WithFabricInterface("qib0").
-				WithFabricInterfacePort(20000).
-				WithPinnedNumaNode(&numaNode0).
-				WithBypassHealthChk(&bypass).
-				WithEnvVars("CRT_TIMEOUT=30").
-				WithLogFile("/tmp/daos_engine.0.log").
-				WithLogMask("WARN"),
-			engine.NewConfig().
-				WithRank(1).
-				WithTargetCount(16).
-				WithHelperStreamCount(6).
-				WithServiceThreadCore(22).
-				WithStorage(
-					storage.NewTierConfig().
-						WithScmMountPoint("/mnt/daos/2").
-						WithScmClass("dcpm").
-						WithScmDeviceList("/dev/pmem1"),
-					storage.NewTierConfig().
-						WithBdevClass("file").
-						WithBdevDeviceList("/tmp/daos-bdev1", "/tmp/daos-bdev2").
-						WithBdevFileSize(16),
-				).
-				WithFabricInterface("qib1").
-				WithFabricInterfacePort(20000).
-				WithPinnedNumaNode(&numaNode1).
-				WithEnvVars("CRT_TIMEOUT=100").
-				WithLogFile("/tmp/daos_engine.1.log").
-				WithLogMask("WARN"),
-		)
+		WithGetNetworkDeviceClass(getDeviceClassStub)
+
+	// add engines explicitly to test functionality applied in WithEngines()
+	constructed.Engines = []*engine.Config{
+		engine.NewConfig().
+			WithSystemName("daos_server").
+			WithSocketDir("./.daos/daos_server").
+			WithRank(0).
+			WithTargetCount(16).
+			WithHelperStreamCount(6).
+			WithServiceThreadCore(0).
+			WithStorage(
+				storage.NewTierConfig().
+					WithScmMountPoint("/mnt/daos/1").
+					WithScmClass("ram").
+					WithScmRamdiskSize(16),
+				storage.NewTierConfig().
+					WithBdevClass("nvme").
+					WithBdevDeviceList("0000:81:00.0"),
+			).
+			WithFabricInterface("qib0").
+			WithFabricInterfacePort(20000).
+			WithFabricProvider("ofi+verbs;ofi_rxm").
+			WithCrtCtxShareAddr(1).
+			WithCrtTimeout(30).
+			WithPinnedNumaNode(&numaNode0).
+			WithBypassHealthChk(&bypass).
+			WithEnvVars("CRT_TIMEOUT=30").
+			WithLogFile("/tmp/daos_engine.0.log").
+			WithLogMask("WARN").
+			WithStorageEnableHotplug(true),
+		engine.NewConfig().
+			WithSystemName("daos_server").
+			WithSocketDir("./.daos/daos_server").
+			WithRank(1).
+			WithTargetCount(16).
+			WithHelperStreamCount(6).
+			WithServiceThreadCore(22).
+			WithStorage(
+				storage.NewTierConfig().
+					WithScmMountPoint("/mnt/daos/2").
+					WithScmClass("dcpm").
+					WithScmDeviceList("/dev/pmem1"),
+				storage.NewTierConfig().
+					WithBdevClass("file").
+					WithBdevDeviceList("/tmp/daos-bdev1", "/tmp/daos-bdev2").
+					WithBdevFileSize(16),
+			).
+			WithFabricInterface("qib1").
+			WithFabricInterfacePort(20000).
+			WithFabricProvider("ofi+verbs;ofi_rxm").
+			WithCrtCtxShareAddr(1).
+			WithCrtTimeout(30).
+			WithPinnedNumaNode(&numaNode1).
+			WithEnvVars("CRT_TIMEOUT=100").
+			WithLogFile("/tmp/daos_engine.1.log").
+			WithLogMask("WARN").
+			WithStorageEnableHotplug(true),
+	}
 	constructed.Path = testFile // just to avoid failing the cmp
 
+	for i := range constructed.Engines {
```

Comment:
this debug will only be printed if the tests fail and will be useful for debug purposes when they do